index.js

```diff
@@ -173,7 +173,14 @@ function compression(options) {
 
       // compression method
       var accept = accepts(req)
-      var method = accept.encoding(['gzip', 'deflate', 'identity'])
+      var method = null;
+
+      // If we support a custom encoding and a custom encoding was requested
+      if (opts.compressor) {
+        method = accept.encoding(Object.keys(opts.compressor));
```

Comment:
How would a used both define a custom encoding (like `br`), but yet still provide `gzip` encoding for clients that do not accept `br`? It looks like users will now need to provide their own `gzip` implementation to achieve this with this PR.